openid-4-verifiable-presentations-1_0.md

```diff
@@ -581,12 +581,24 @@ The following is a non-normative example of the Authorization Response that is s
     state=eyJhb...6-sVA
 ```
 
-If the Response Endpoint has successfully processed the request, it MUST respond with HTTPS status code 200. 
+The following is a non-normative example of an Authorization Error Response that is sent as an HTTP POST request to the Verifier's Response Endpoint:
+
+```
+  POST /post HTTP/1.1
+    Host: client.example.org
+    Content-Type: application/x-www-form-urlencoded
+
+    error=...&
+    error_description=...&
+    state=eyJhb...6-sVA
```

Comment:
I believe `state` becomes a MUST in this case. Otherwise, how we would distinguish one error response from another.

Reply:
`state` is REQUIRED as per OAuth2 if provided in the Authz Request. We have that reference in OID4VP already. So, in that case `state` is REQUIRED.

> Other parameters, such as state or code (from [[RFC6749](https://openid.github.io/OpenID4VP/openid-4-verifiable-presentations-wg-draft.html#RFC6749)]), or id_token (from [[OpenID.Core](https://openid.github.io/OpenID4VP/openid-4-verifiable-presentations-wg-draft.html#OpenID.Core)]), and iss (from [[RFC9207](https://openid.github.io/OpenID4VP/openid-4-verifiable-presentations-wg-draft.html#RFC9207)]) MAY be included in the response as defined in the respective specifications.

Reply:
I (perhaps incorrectly) assumed @adeinega was talking about requiring `state` in the request. I agree the situation in the response is already clear. I think we could be clearer about the use of state in requests in the direct post case at least, I might raise a separate issue for that.

Reply:
@jogu, I was talking about `state` in the response and specifically, about this type of the error response.

@awoie, yes, you are right, `state` is required in the error response if it was presented in the authz request (1), and the authorization server redirects a user to the client (2) per https://datatracker.ietf.org/doc/html/rfc6749#section-4.1.2.1, but for other cases, RFC 6749 doesn't even mention it, take a quick look at https://datatracker.ietf.org/doc/html/rfc6749#section-5.2.

Reply:
>  for other cases, RFC 6749 doesn't even mention it, take a quick look at https://datatracker.ietf.org/doc/html/rfc6749#section-5.2.

This is the token endpoint response section. state is not required there, that's why it's not mentioned. As you say, the authorization error response section (which is the only one relevant to this discussion) does mention state.